compiler/packages/babel-plugin-react-compiler/src/HIR/HIR.ts

```diff
@@ -1507,7 +1507,17 @@ export type ReactiveScopeDeclaration = {
   scope: ReactiveScope; // the scope in which the variable was originally declared
 };
 
-export type DependencyPathEntry = {property: string; optional: boolean};
+const opaquePropertyLiteral = Symbol();
+export type PropertyLiteral = (string | number) & {
+  [opaquePropertyLiteral]: 'PropertyLiteral';
+};
+export function makePropertyLiteral(value: string | number): PropertyLiteral {
+  return value as PropertyLiteral;
+}
```

Comment:
This is the primary change: all 'static' instructions (PropertyLoad, PropertyStore, and PropertyDelete) can now operate on either string or numeric indices